weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -252,17 +249,17 @@ def calls_query_stream(
         cq = CallsQuery(
             project_id=req.project_id, include_costs=req.include_costs or False
         )
-
-        # TODO (Perf): By allowing a sub-selection of columns
-        # we will gain increased performance by not having to
-        # fetch all columns from the database. Currently all use
-        # cases call for every column to be fetched, so we have not
-        # implemented this yet.
         columns = all_call_select_columns
+        if req.columns:
```

Comment:
it would be nice to add a comment here saying that while `columns` is user-provided, the query builder itself enforces the correct column names and prevents injection. Without such context, this looks scary to me (: